x/mm2/module.go

```diff
@@ -0,0 +1,257 @@
+	if err := cdc.UnmarshalJSON(bz, &gs); err != nil {
+		return err
+	}
+
+	return gs.ValidateBasic()
+}
+
+// RegisterInvariants registers the invariants of the marketmap module. If an invariant
+// deviates from its predicted value, the InvariantRegistry triggers appropriate
+// logic (most often the chain will be halted). No invariants exist for the marketmap module.
+func (AppModule) RegisterInvariants(_ sdk.InvariantRegistry) {}
+
+// AppModule is the actual app module for x/marketmap.
+type AppModule struct {
+	AppModuleBasic
+	k *keeper.Keeper
+}
+
+// BeginBlock is a no-op for x/marketmap.
+func (am AppModule) BeginBlock(_ context.Context) error {
+	return nil
+}
+
+// EndBlock is a no-op for x/marketmap.
+func (am AppModule) EndBlock(_ context.Context) error {
+	return nil
+}
+
+// InitGenesis performs the genesis initialization for the x/marketmap module. It determines the
+// genesis state to initialize from via a json-encoded genesis-state. This method returns no validator set updates.
+// This method panics on any errors.
+func (am AppModule) InitGenesis(ctx sdk.Context, cdc codec.JSONCodec, bz json.RawMessage) {
+	// unmarshal genesis-state (panic on errors)
+	var gs types.GenesisState
+	cdc.MustUnmarshalJSON(bz, &gs)
+
+	// initialize genesis
+	am.k.InitGenesis(ctx, gs)
+}
+
+// ExportGenesis returns the oracle module's exported genesis state as raw
+// JSON bytes. This method panics on any error.
+func (am AppModule) ExportGenesis(ctx sdk.Context, cdc codec.JSONCodec) json.RawMessage {
+	gs := am.k.ExportGenesis(ctx)
+	return cdc.MustMarshalJSON(gs)
+}
+
+// RegisterServices registers the module's services with the app's module configurator.
+func (am AppModule) RegisterServices(cfc module.Configurator) {
+	// register MsgServer
+	types.RegisterMsgServer(cfc.MsgServer(), keeper.NewMsgServer(am.k))
+
+	// register Query Service
+	types.RegisterQueryServer(cfc.QueryServer(), keeper.NewQueryServer(am.k))
+}
+
+// ConsensusVersion implements AppModule/ConsensusVersion.
+func (AppModule) ConsensusVersion() uint64 { return ConsensusVersion }
+
+// IsOnePerModuleType implements the depinject.OnePerModuleType interface.
+func (am AppModule) IsOnePerModuleType() {}
+
+// IsAppModule implements the appmodule.AppModule interface. It is a no-op.
+func (am AppModule) IsAppModule() {}
+
+// NewAppModule constructs a new application module for the x/marketmap module.
+func NewAppModule(cdc codec.Codec, k *keeper.Keeper) AppModule {
+	return AppModule{
+		AppModuleBasic: AppModuleBasic{
+			cdc: cdc,
+		},
+		k: k,
+	}
+}
+
+/*
+
+	Dep-Inject
+
+*/
+
+var _ depinject.OnePerModuleType = AppModule{}
+
+func init() {
+	appmodule.Register(
+		&marketmapmodulev1.Module{},
+		appmodule.Provide(ProvideModule),
+		appmodule.Invoke(InvokeSetMarketMapHooks),
+	)
+}
+
+// Inputs contains the dependencies required for module construction.
+type Inputs struct {
+	depinject.In
+
+	// module dependencies
+	Config       *marketmapmodulev1.Module
```

Comment:
We're using the existing marketmap to construct the new marketmap? Is that correct?

Reply:
no - if you look at the import path - this is also from the new proto

Reply:
Ah I see. Just a confusing import name I guess. :O